runner/repositories.go

```diff
@@ -63,11 +60,20 @@ func (r *Runner) CreateRepository(ctx context.Context, param params.CreateRepoPa
 		}
 	}()
 
-	poolMgr, err := r.loadRepoPoolManager(repo)
+	internalCfg, err := r.getInternalConfig(repo.CredentialsName)
+	if err != nil {
+		return params.Repository{}, errors.Wrap(err, "fetching internal config")
+	}
+	repo.Internal = internalCfg
+
+	poolMgr, err := r.poolManagerCtrl.CreateRepoPoolManager(r.ctx, repo, r.providers, r.store)
+	if err != nil {
+		return params.Repository{}, errors.Wrap(err, "creating repo pool manager")
+	}
 	if err := poolMgr.Start(); err != nil {
-		return params.Repository{}, errors.Wrap(err, "starting pool manager")
+		r.poolManagerCtrl.DeleteRepoPoolManager(repo)
```

Comment:
If this fails, we need to know.

Reply:
I will check error here and wrap it further, if needed.